subsys/net/lib/tftp/tftp_client.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2020 InnBlue
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Generic header files required by the TFTPC Module. */
+#include <zephyr.h>
+#include <errno.h>
+#include <net/socket.h>
+
+/* Defines for creating static arrays for TFTP communication. */
+#define TFTP_MAX_FILENAME_SIZE   CONFIG_TFTPC_MAX_FILENAME_SIZE
+#define TFTP_HEADER_SIZE         4
+#define TFTP_BLOCK_SIZE          512
+#define TFTP_MAX_MODE_SIZE       8
+#define TFTP_REQ_RETX            CONFIG_TFTPC_REQUEST_RETRANSMITS
+
+/* Don't allow file names of more than "256" chars. */
+#if (TFTP_MAX_FILENAME_SIZE > 256)
+#error "TFTPC_MAX_FILENAME_SIZE cannot be greater than 256"
+#endif /* #if (TFTP_MAX_FILENAME_SIZE > 256) */
+
+/* Maximum amount of data that can be received in a single go ! */
+#define TFTPC_MAX_BUF_SIZE       (TFTP_BLOCK_SIZE + TFTP_HEADER_SIZE)
+
+/* TFTP Opcodes. */
+#define READ_REQUEST             0x1
+#define WRITE_REQUEST            0x2
+#define DATA_OPCODE              0x3
+#define ACK_OPCODE               0x4
+#define ERROR_OPCODE             0x5
+
+#define RECV_DATA_SIZE()         (tftpc_buffer_size - TFTP_HEADER_SIZE)
+
+/* Name: get_u16
+ * Description: This function gets u16_t buffer into a
+ * u8_t buffer array.
+ */
+static inline u16_t get_u16(u8_t *buf)
```

Comment:
could you use instead sys_get_be16() from include/sys/byteorder.h

Reply:
Great suggestion, Thanks.. I thought that such things would be available within net but never really looked..